src/container.ts

```diff
@@ -428,6 +429,12 @@ export function createContainer<T extends object = any, U extends object = any>(
     return null
   }
 
+  function getDefaultArgs() {
```

Comment:
I don't think this will work if using `container.build(asClass(...).withDefaultArgs())`.

Reply:
Ah no, it does not. Is there a smarter way to get the parent resolver options?

Reply:
Maybe we can pass in `allowUnregistered: true` in the resolver the same way we do for `CLASSIC`, and in the case it returns `undefined` (or `null`, can't remember), use the default?

Reply:
But then it would only work with the default args set in the constructor? In that case I think it would make more sense to force default args to always be set in the constructor and change the API to be something like

```
asClass(...).useDefaultArgs({foo: true})
```

and then specify what `foo` should actually be in the constructor/function instead of allowing it to be specified in the registration.

Reply:
What do you mean by "constructor" here? Are you referring to `asClass()...` chaining to configure the default args, or are you talking about an actual class constructor?

It should work with inline resolver options since we merge the options here.

https://github.com/jeffijoe/awilix/blob/4c6b615d197f52dafae19cc91e0b8664a4c2317d/src/resolvers.ts#L160

Reply:
I mean the actual constructor.

The issue is that at the point we're calling `getDefaultArgs()` we're attempting to resolve the dependency called `unregistered` in the tests. However the setting for `defaultArgs` is set on the parent of that module.

Like you say it would be possible to pass `allowUnregistered` but in this case there is still no possibility to retrieve the config of the parent (As far as I can find). I've added a test for the `build()` function, and in that case I don't see any way to get `defaultArgs` from the class called `DefaultArgs` while trying to resolve `unregistered`.

Reply:
And unless I'm missing something passing `allowUnregistered` is not an option, since the `resolve()` function is called from the `get()` that's triggered from the line `this.unregistered = cradle.unregistered` in the class constructor.

Reply:
Yeah, I agree that's adding a bit too much complexity.

What is the reason that the `resolutionStack` does not exist when using `build`? Is it possible to construct it in this case in the same way as it's constructed when using the container in a "regular" way?

Reply:
Because `build` just takes a resolver and constructs it, it doesn't register it, and only registered things go in the resolution stack (it's used to detect and print circular refs)

Reply:
I see, that makes sense. Would it be an acceptable compromise that `defaultArgs` doesn't work with `build()` provided this is clearly documented?

Reply:
I'd prefer covering that case as well. We would have to measure how passing down the default args map through the resolve chain affects performance.

Also, I'm not sure if you are aware, but I believe you can do something like this:

```js
asClass(MyClass).inject(c => ({ maybeValue: c.resolve('maybeValue', { allowUnregistered: true }))
```

Reply:
Typo. 😄  You used `unresolved` instead of `unregistered`.

Reply:
That seems to work in the context of the tests, however when I try it in our actual code the `get()` call to the `cradle` immediately goes to `container.ts` instead of `resolvers.ts`. I would presume this is due to some config or setup, do you have any idea off the top of your head what config might make it skip the locally injected variables?

Reply:
Yeah, it seems the `loadModules` isn't picking up the right symbol in my actual project. I'm looking at the code for `registerDescriptor` and the `moduleDescriptor.value` has an attribute for Awilix Resolver Config:
![image](https://user-images.githubusercontent.com/1245031/142030005-3f21a660-11cd-4dad-b02d-60609d18b086.png)

However `moduleDescriptor.value[resolvers_1.RESOLVER]` (This is in the JSified TS code) still returns `undefined` 🤔  It's a bit far from the topic of the PR but is this something you've encountered before? Somehow it seems to be two different Symbol instances.

Reply:
I haven’t because I don’t even like the in-line resolver feature 😅 IMO the classes themselves should be completely free of any DI gunk, and any class that requires special treatment in DI setup I just configure manually in the container. I’m not adding new classes to my project every 30 seconds so I don’t mind having to manually register dependencies in the container.

Reply:
Are you maybe using 2 different versions of Awilix at the same time?

Reply:
Ah I figured it out, the service has both a direct and indirect dependency on Awilix. Simply doing `import { RESOLVER } from 'awilix'` will import the "wrong" symbol as it imports the one from the direct dependency instead from the indirect one (Which is the one that does the module loading later on). I actually think the `injector` might do the trick, so I may convert this PR into simply updating the docs with instructions on how to do this!